crates/ruff_python_formatter/resources/test/fixtures/ruff/expression/join_implicit_concatenated_string_preserve.py

```diff
@@ -11,3 +11,12 @@
 
 # Already invalid Pre Python 312
 f"{'Hy "User"'}" f'{"Hy 'User'"}'
+
+
+# Regression tests for https://github.com/astral-sh/ruff/issues/15514
+params = {}
+string = "this is my string with " f'"{params.get("mine")}"'
+string = f'"{params.get("mine")} ' f"with {'nested single quoted string'}"
+string = f"{'''inner ' '''}" f'{"""inner " """}'
+string = f"{10 + len('bar')=}" f"{10 + len('bar')=}"
+string = f"{10 + len('bar')=}" f'{10 + len("bar")=}'
```

Comment:
We might want to add some test cases which includes format spec.

Is the following correct?
```diff
 params = {}
-string = "this is my string with " f'"{params.get("mine")=:{"xxx"}}"'
+string = f'this is my string with "{params.get("mine")=:{"xxx"}}"'
```

with quote-style `preserve` and < 3.12 Python.

Or, is there no way to preserve the quote style here?

Reply:
This doesn't seem to be relevant to this PR though.

Reply:
There's no strictly correct answer when it comes to preserve and joining implicit concatenated strings if not all parts use the same quotes. You have to choose which quotes you want to preserve and in this example, the formatter decides to preserve the quotes from the second part because the string can't be joined otherwise.

Reply:
That sounds reasonable, thanks